spanner/statement_test.go

```diff
@@ -159,6 +159,12 @@ func TestConvertParams(t *testing.T) {
 			listProto(listProto(intProto(10)), listProto(intProto(20))),
 			listType(structType(mkField("field", intType()))),
 		},
+		// Untyped null
+		{
+			nil,
+			nullProto(),
+			nil,
```

Comment:
Just curious: would this work as it expects a *sppb.Type but we give nothing. E.g., 

```
func numericType() *sppb.Type {
	return &sppb.Type{Code: sppb.TypeCode_NUMERIC}
}
```

Reply:
In this test it would fail, as the test will verify that the param type is not set (or set to nil). That comparison will fail if we set it to `TypeCode_NUMERIC`. If we were to do something like that on production, it will also fail. I tried it out with the integration test, and it will refuse to insert a `NULL` value with type `NUMERIC` into a `STRING` column:

```
failed to insert values using DML: spanner: code = "InvalidArgument", desc = "Value has type NUMERIC which cannot be inserted into column String, which has type STRING [at 1:50]\\nINSERT INTO Types (RowId, `String`) VALUES (@id, @value)
```